salt/modules/boto3_sns.py

```diff
@@ -116,8 +116,8 @@ def describe_topic(name, region=None, key=None, keyid=None, profile=None):
                 arn, region=region, key=key, keyid=keyid, profile=profile
             )
             # Grab extended attributes for the above subscriptions
-            for sub in range(len(ret["Subscriptions"])):
-                sub_arn = ret["Subscriptions"][sub]["SubscriptionArn"]
+            for sub in ret["Subscriptions"]:
+                sub_arn = sub["SubscriptionArn"]
```

Comment:
Isn't `sub` here just the key?

Reply:
`for sub in ret["Subscriptions"].values():`

Reply:
Subscriptions is a list not dict